systems/data-plan/package/pkg/db/package_repo.go

```diff
@@ -22,10 +22,16 @@ func NewPackageRepo(db sql.Db) *packageRepo {
 	}
 }
 
-func (r *packageRepo) Add(_package *Package, nestedFunc ...func() error) error {
-	err := r.Db.ExecuteInTransaction(func(tx *gorm.DB) *gorm.DB {
-		return tx.Create(_package)
-	}, nestedFunc...)
+func (r *packageRepo) Add(_package *Package) error {
+	err := r.Db.GetGormDb().Transaction(func(tx *gorm.DB) error {
+		result := tx.Create(_package)
+
+		if result.Error != nil {
```

Comment:
```go
if result.Error != nil {
			return result.Error
		}

		return nil
```
Then we can just  `return restult.Error`

Reply:
Addressed